aws-synthetics-canary/src/main/java/com/amazon/synthetics/canary/CreateHandler.java

```diff
@@ -12,87 +11,94 @@
-                        .build() :
-                callbackContext;
-        syntheticsClient = ClientBuilder.getClient();
+            CallbackContext.builder().build() :
+            callbackContext;
+        SyntheticsClient syntheticsClient = ClientBuilder.getClient();
 
-        // This Lambda will continually be re-invoked with the current state of the instance, finally succeeding when state stabilizes.
-        return createCanaryAndUpdateProgress(model, currentContext, proxy, request, syntheticsClient);
-    }
-
-    private ProgressEvent<ResourceModel, CallbackContext> createCanaryAndUpdateProgress(final ResourceModel model,
-                                                                                        final CallbackContext callbackContext,
-                                                                                        final AmazonWebServicesClientProxy proxy,
-                                                                                        final ResourceHandlerRequest<ResourceModel> request,
-                                                                                        final SyntheticsClient syntheticsClient) {
-        // Create Canary
-        if (!callbackContext.isCanaryCreationStarted()) {
-            return createCanary(model, callbackContext, proxy, request, syntheticsClient);
-        }
+        logger.log(String.format("[CREATE] Create handler called for canary %s. RetryKey = %s and RemainingRetryCount = %d.",
+            model.getName(), currentContext.getRetryKey(), currentContext.getRemainingRetryCount()));
 
-        //Canary creation started. Check for stabilization.
-        if (!callbackContext.isCanaryCreationStablized()) {
-            return updateCanaryCreationProgress(model, callbackContext, proxy, request, syntheticsClient);
-        }
+        if (!currentContext.isCanaryCreationStarted()) {
+            // Creation has yet to begin
 
-        // Canary has been successfully created. Check if it needs to be started or return SUCCESS
-        if (model.getStartCanaryAfterCreation()) {
-            if (!callbackContext.isCanaryStartStarted()) {
-                return startCanary(model, callbackContext, proxy, request, syntheticsClient);
-            }
+            logger.log(String.format("[CREATE] Creating canary %s.", model.getName()));
+            currentContext.setCanaryCreationStarted(true);
 
-            // Canary has been started. Wait until it stabilizes
-            if (!callbackContext.isCanaryStartStablized()) {
-                return updateCanaryStartProgress(model, callbackContext, proxy, request, syntheticsClient);
-            }
+            return createCanary(model, currentContext, proxy, request, syntheticsClient);
+        }
 
-            // Canary has been started. Wait until it stabilizes
+        Canary canary = CanaryHelper.getCanaryOrThrow(proxy, syntheticsClient, model);
+        if (canary.status().state() == CanaryState.CREATING) {
```

Comment:
Consider using switch

Reply:
Also..might be a good idea to modularize this? Having one method with nested if else is generally not a good idea